tableauserverclient/server/endpoint/auth_endpoint.py

```diff
@@ -41,6 +41,9 @@ def sign_in(self, auth_req):
 
     def sign_out(self):
         url = "{0}/{1}".format(self.baseurl, 'signout')
+        # If there are no auth tokens you're already signed out. No-op
+        if not self.parent_srv._auth_token:
```

Comment:
This feels like it should be a method on the server object to indicate if it is signed in. Looking at what you wrote, I understand that there is no "private" in python but we should honor things when they make sense. A method on the server liked signed_in() which returns a boolean would feel cleaner.